runtime/src/staking.rs

```diff
@@ -0,0 +1,101 @@
+use crate::Perbill;
+
+use codec::{Decode, EncodeLike};
+use frame_support::{traits::Currency, dispatch::{DispatchResultWithPostInfo, Vec}};
+use frame_system::{Config as SysConfig, RawOrigin};
+use pallet_staking::{Pallet as StakingPallet, BalanceOf};
+use pallet_utxo::staking::StakingHelper;
+use sp_core::{H256, sp_std::vec};
+use sp_runtime::traits::StaticLookup;
+
+type StakeAccountId<T> =  <T as SysConfig>::AccountId;
+type LookupSourceOf<T> = <<T as SysConfig>::Lookup as StaticLookup>::Source;
+
+pub struct StakeOps<T>(sp_core::sp_std::marker::PhantomData<T>);
+impl <T: pallet_staking::Config + pallet_utxo::Config + pallet_session::Config> StakingHelper<T::AccountId> for StakeOps<T>
+    where StakeAccountId<T>: From<[u8; 32]>,
+          BalanceOf<T>: From<u128>
+{
+    fn get_account_id(pub_key: &H256) -> StakeAccountId<T> {
+        pub_key.0.into()
+    }
+
+    fn stake(stash_account: &StakeAccountId<T>, controller_account: &StakeAccountId<T>, session_key: &mut Vec<u8>, value: u128) -> DispatchResultWithPostInfo {
+        let controller_lookup: LookupSourceOf<T> = T::Lookup::unlookup(controller_account.clone());
+        let reward_destination = pallet_staking::RewardDestination::Staked;
+
+        // bond the funds
+        StakingPallet::<T>::bond(
+            RawOrigin::Signed(stash_account.clone()).into(),
+            controller_lookup,
+            value.into(),
+            reward_destination
+        )?;
+
+        let rotate_keys = sp_core::Bytes::from(session_key.to_vec());
+        // session keys
+        let sesh_key = <T as pallet_session::Config>::Keys::decode(&mut &rotate_keys[..]).expect("SessionKeys decoded successfully");
+        pallet_session::Pallet::<T>::set_keys(
+            RawOrigin::Signed(controller_account.clone()).into(),
+            sesh_key,
+            vec![]
+        )?;
+
+        let validator_prefs = pallet_staking::ValidatorPrefs {
+            commission: Perbill::from_percent(0),
+            ..Default::default()
+        };
+
+        // validate
+        StakingPallet::<T>::validate(
```

Comment:
I think code in this file can be simplified to use the return value of these `StakingPallet` calls. Like here the semicolon and `Ok(().into())` could be removed because `validate` seems to return `DispatchResultWithPostInfo` as well